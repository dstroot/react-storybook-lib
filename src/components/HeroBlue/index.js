import React from "react";
import PropTypes from "prop-types";

const HeroBlue = ({ align, image, tagline, text, attribution }) => {
  const heroStyle = {
    /* Use "linear-gradient" to add a darken background effect to the image (photographer.jpg). This will make the text easier to read */
    // backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url("photographer.jpg")`,
    backgroundImage: `url(${image})`,

    /* Set a specific height */
    // height: "45vh",
    minHeight: "40vw",
    maxHeight: "50vh",

    /* Position and center the image to scale nicely on all screens */
    backgroundPosition: "center",
    backgroundRepeat: "no-repeat",
    backgroundSize: "cover",
    position: "relative"
  };

  const divOverlay = {
    marginTop: "-12vh"
  };

  const textLarge = {
    fontSize: `calc(44px + 1.5vw + 0.5vmin)`,
    lineHeight: "1.2",
    fontWeight: "300"
  };

  const textSmall = {
    fontSize: `calc(20px + 1.5vw + 0.5vmin)`,
    lineHeight: "1.2",
    fontWeight: "300"
  };

  return (
    <>
      <div className="hero text-white" style={heroStyle} />
      <div className="row" style={divOverlay}>
        <div className="col-md-10 offset-md-1 p-3 text-center text-white bg-primary">
          <p style={textLarge}>{tagline || ""}</p>
          <p style={textSmall}>{text || ""}</p>
        </div>
      </div>
    </>
  );
};

HeroBlue.propTypes = {
  /** The background image URL  */
  image: PropTypes.string.isRequired,
  /** The tagline */
  tagline: PropTypes.string.isRequired,
  /** Additional text (options) */
  text: PropTypes.string
};

export default HeroBlue;
