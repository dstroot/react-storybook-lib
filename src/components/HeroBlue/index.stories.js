import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import HeroBlue from '../HeroBlue';

// images
// import image1 from './media/1.jpg';
import image2 from './media/2.jpg';
import image3 from './media/3.jpg';
import image4 from './media/startup.jpg';
import image5 from './media/home.jpg';

storiesOf('HeroBlue', module)
  .add('show HeroBlue image 1', () => (
    <HeroBlue
      align="right"
      image="https://www.pacificlife.com/content/dam/paclife/crp/images/brand/Brand_hero.jpg"
      tagline="You. Protected."
      text="Our mission is protecting your financial future. For 150 years, we’ve fulfilled that promise."
    />
  ))
  .add('show HeroBlue image 2', () => (
    <HeroBlue
      align="left"
      image={image2}
      tagline="You. Protected."
      text="Our mission is protecting your financial future. For 150 years, we’ve fulfilled that promise."
    />
  ))
  .add('show HeroBlue image 3', () => (
    <HeroBlue
      image={image3}
      align="center"
      tagline="You. Protected."
      text="Our mission is protecting your financial future. For 150 years, we’ve fulfilled that promise."
    />
  ))
  .add('show HeroBlue image 4', () => (
    <HeroBlue
      image={image4}
      align="right"
      tagline="You. Protected."
      text="Our mission is protecting your financial future. For 150 years, we’ve fulfilled that promise."
    />
  ))
  .add('show HeroBlue image 5', () => (
    <HeroBlue
      image={image5}
      align="center"
      tagline="You. Protected."
      text="Our mission is protecting your financial future. For 150 years, we’ve fulfilled that promise."
    />
  ));
