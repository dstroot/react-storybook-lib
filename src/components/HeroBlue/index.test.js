import React from "react";
import renderer from "react-test-renderer";

import HeroBlue from "../HeroBlue";
import image from "../../components/HeroBlue/media/startup.jpg";

test("HeroBlue renders some content", () => {
  const component = renderer.create(
    <HeroBlue image={image} tagline="It's Miller Time!" />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
