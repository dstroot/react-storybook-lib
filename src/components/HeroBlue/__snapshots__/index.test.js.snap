// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBlue renders some content 1`] = `
Array [
  <div
    className="hero text-white"
    style={
      Object {
        "backgroundImage": "url(startup.jpg)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "maxHeight": "50vh",
        "position": "relative",
      }
    }
  />,
  <div
    className="row"
    style={
      Object {
        "marginTop": "-12vh",
      }
    }
  >
    <div
      className="col-md-10 offset-md-1 p-3 text-center text-white bg-primary"
    >
      <p
        style={
          Object {
            "fontSize": "calc(44px + 1.5vw + 0.5vmin)",
            "fontWeight": "300",
            "lineHeight": "1.2",
          }
        }
      >
        
      </p>
      <p
        style={
          Object {
            "fontSize": "calc(20px + 1.5vw + 0.5vmin)",
            "fontWeight": "300",
            "lineHeight": "1.2",
          }
        }
      >
        
      </p>
    </div>
  </div>,
]
`;
